src/components/NSLookupTool/NSLookupTool.tsx

```diff
@@ -60,40 +76,36 @@ export function NSLookup() {
             } else {
                 handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
             }
-            // Clear the child process pid reference
             setPid("");
-            // Cancel the Loading Overlay
             setLoading(false);
-
-            // Allow Saving as the output is finalised
             setAllowSave(true);
             setHasSaved(false);
         },
-        [handleProcessData] // Dependency on the handleProcessData callback
+        [handleProcessData]
     );
-    /**
-     * onSubmit: Handler function that is triggered when the form is submitted.
-     * It prepares the arguments and initiates the execution of the `bed` command.
-     * Upon successful execution, it updates the state with the process PID and output.
-     * If an error occurs during the command execution, it updates the output with the error message.
-     * @param {FormValues} values - An object containing the form input values.
-     */
-
-    // Actions taken after saving the output
+
+    // Updating the state of Saving the output
     const handleSaveComplete = () => {
-        // Indicating that the file has saved which is passed
-        // back into SaveOutputToTextFile to inform the user
         setHasSaved(true);
         setAllowSave(false);
     };
 
-    const onSubmit = (values: FormValues) => {
-        // Disallow saving until the tool's execution is complete
-        setAllowSave(false);
+    // Handling the submitted form
+    const onSubmit = (values: FormValuesType) => {
+        const ipAddress = values.ipAddress.trim();
+        if (!ipAddress) {
+            setOutput("Error: No IP address or hostname provided.");
+            return;
+        }
 
+        setAllowSave(false);
         setLoading(true);
-        const args = [values.ipaddress];
-        CommandHelper.runCommandGetPidAndOutput("nslookup", args, handleProcessData, handleProcessTermination)
+ hathimsh
```

Comment:
Hi @hathimsh I think you will need to remove this to pass the CI checks. 
```suggestion

```

Reply:
done the changes